### PR TITLE
Run lint+tests on push, block merge

### PR DESCRIPTION
**Pull request by Invertisment**

### CODE REVIEW

File: .github/workflows/test-after-push.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Run tests on push
+on: push
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
```
Comment (robert-zaremba) [inline .github/workflows/test-after-push.yaml:8]: what is `checkout@v1`? And why there is `@1`?

Comment (Invertisment) [reply .github/workflows/test-after-push.yaml:8]: This checks out the repository. It didn't work without this line. 
This is what happens when this checkout is not executed:
https://github.com/scale-it/algorand-builder/runs/922469012?check_suite_focus=true

Comment (Invertisment) [reply .github/workflows/test-after-push.yaml:8]: This is what it does:
https://github.com/marketplace/actions/checkout

Comment (robert-zaremba) [reply .github/workflows/test-after-push.yaml:8]: let's update to `v2` if possible.

Comment (Invertisment) [reply .github/workflows/test-after-push.yaml]: updated

**Pull request merged**